app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,141 @@ angular.module('confRegistrationWebApp')
     $scope.hasPendingCheckPayment = function(payments){
       return _.some(payments, { paymentType: 'CHECK', status: 'PENDING' });
     };
+
+    ////// START EVENT-433 //////
+
+    // Get spouse registration
+    // End point: getSpouseRegistration = /conferences/conferenceId/registrations/spouse
+    ///// TEST DATA change end point url below to: /conferences/conferenceId/registrations/current /////
+    $http.get('conferences/' + conference.id + '/registrations/spouse')
+      .then(function(response) {
+        if (response.status === 200) {
+          $scope.spouseRegistration = response.data;
+
+          ///// TEST DATA /////
+          // $scope.spouseRegistration.completed = true;
+          // $scope.spouseRegistration.registrants[0].email = 'janeworner3@hotmail.com';
+          // $scope.spouseRegistration.registrants[0].firstName = 'Jane';
+          // $scope.spouseRegistration.registrants[0].lastName = 'Worner3';
+          ///// TEST DATA ////
+
+        }
+        else {
+          $scope.spouseRegistration = {};
+          console.log('No current and/or spouse registration found for user with ID.');
+        }
+      }).catch(function(response) {
+      $scope.spouseRegistration = {};
+      console.log('No current and/or spouse registration found for user with ID. Status = ' + response.status);
+      console.log(response.data);
+    })
+      .finally(function() {
+        checkSpouseRegistration();
+      });
+
+    //Parameters needed to retrieve all current registrants for conference.id
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    //then check if spouse is registered and display 'Spouse Registration' section if needed.
+    var checkSpouseRegistration = function() { // jshint ignore:line
+      $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
```

Comment:
@canac It was based off of 2 things.  1. the requirement, and 2. what I saw existed in memory on the $scope.currentRegistration object when setting a breakpoint.

Per #1:
_A. If spouse is not included in that registration, then show another separate panel on the confirmation page, indicating that their spouse has already registered and would they like to add their registration to their spouses?_
Based on this requirement I interpreted "If spouse I not included in that registration..." to refer to the list of existing registrants who had already registered for that event.  So if an event had 4 registrants, one of which was the spouse, we would need to check for this.  And only way would be to check full list of registrants already registered for the event.

Per #2:
When setting a breakpoint and debugging I noticed that upon arriving to the page with the 'Confirm' button that only 1 registrant is listed in the $scope.currentRegistration object.  The current logged in registrant.  So if my name was "John Doe" and I was registering for an event for the first time, then upon arriving to the page with the 'Confirm' button I would be the only registrant listed in this object.  So if my wife's name was "Jane Doe" which was returned as my spouse and listed in $scope.spouseRegistration then there would never be a match in emails because "John Doe's" email would always be different than "Jane Doe's" since both $scope.currentRegistration and $scope.spouseRegistration both only list 1 registrant.

The only way I found to do this was to get a complete list of registrants.

Does this make sense?

It might be better to have a quick call  651-964-9873